src/pudl_archiver/archivers/classes.py

```diff
@@ -100,20 +101,45 @@ async def download_zipfile(
         # If it makes it here that means it couldn't download a valid zipfile
         raise RuntimeError(f"Failed to download valid zipfile from {url}")
 
-    async def download_file(self, url: str, file: Path | io.BytesIO, **kwargs):
+    async def _get_with_retries(
+        self, url, retry_count: int = 5, retry_base_s: int = 1, **kwargs
+    ):
+        for try_count in range(1, retry_count + 1):
+            # try count is 1 indexed for logging clarity
+            try:
+                self.logger.info(f"GET {url} (try #{try_count})")
+                response = await self.session.get(url, **kwargs)
+                break
+            except aiohttp.ClientError as e:
+                if try_count == retry_count:
+                    raise e
+                retry_delay_s = retry_base_s * 2**try_count
+                self.logger.info(
+                    f"ClientError while getting {url} (try #{try_count}, retry in {retry_delay_s}s): {e}"
+                )
+                time.sleep(retry_delay_s)
```

Comment:
That makes sense, though if we have a bunch of concurrent tasks hitting the same endpoint we might just have everyone hitting the same rate limit anyways. Regardless of that, it would be easier to reason about `asyncio.sleep` I think so I'll swap that over.